src/crimson/osd/heartbeat.cc

```diff
@@ -309,6 +310,29 @@ seastar::future<> Heartbeat::handle_you_died()
   return seastar::now();
 }
 
+void Heartbeat::heartbeat_check()
+{
+  bool failure = false;
```

Comment:
```suggestion
  unsigned failures = 0;
```

it'd help with debugging. moreover, i think it's simpler to just remove `failure_queue` member variable and just return the identified osds, and pass them to `send_heartbeats()`.

Reply:
I got your point:-) Will try that.

Reply:
Um...By send_heartbeats(), what you actually mean is send_failures(), right? Or, are you suggesting merging the send_failures() with send_heartbeats()?

Reply:
sorry, i meant to say `send_failures()`.